### ISSUE TITLE
Windows inside Tabs

### ISSUE THREAD

**Issue by IceLuna** [ocornut/imgui#6221]
**Version/Branch of Dear ImGui:**

Version: 1.89.4 WIP
Branch: docking

**My Issue/Question:**

Hi!
I want to make UI like in unreal engine and my question is: is there a way to dock windows to a tab?
On a screenshot below you can see a texture-viewer tab (called `T_Test`) and it has docked windows (Viewport, Details, etc).
Also these docked windows are moved along with a texture-viewer tab
How can I achieve that? Couldn't find anything like this in issues and in `ImGui::ShowDemoWindow()`
I hope I didn't miss anything :)
Thanks!

**Screenshots/Video**

![ue_example](https://user-images.githubusercontent.com/48557794/222919184-877ff7c4-33c7-42ad-8182-dc813608a431.png)

**Comment by ocornut** [comment]
If you create a `DockSpace()` inside your window you can dock other windows into it.
See Demo->Examples->Dockspace and Demo->Examples->Documents.

**Comment by IceLuna** [comment]
Yes, I've tried it but I don't see a way to recreate UE's behavior.
I did something like this
![image](https://user-images.githubusercontent.com/48557794/223210268-6f185880-f8b3-486e-9f16-c546822b37ea.png)

I have `Viewport` and `Details` windows like in UE. But the key that is missing is a tab containing both windows (in UE-screenshot it's called `T_Test`). And I can't dock these two windows to something else without undocking them.
I guess I need some kind of Parent/Upper window that will carry `Viewport` and `Details` windows along with it.

**Comment by SShikhali** [comment]
Sorry, but I can't seem to understand it and get it right. I've tried playing with `Demo->Examples->Dockspace` and `Demo->Examples->Documents` and couldn't get what I want. This is what I have right now:

![image](https://user-images.githubusercontent.com/99796443/224504941-be7b8e74-2756-4a9c-867a-d097ee9cbc5e.png)

```
ImGui::Begin("Tab/Parent window");
ImGui::DockSpace(dockspace_id1, ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_KeepAliveOnly);
ImGui::End();
		
ImGui::Begin("MyViewport");
ImGui::DockSpace(dockspace_id2, ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_KeepAliveOnly);
ImGui::End();
		
ImGui::Begin("Details");
ImGui::DockSpace(dockspace_id3, ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_KeepAliveOnly);
ImGui::End();
```

My goal result is something like this
![image2](https://user-images.githubusercontent.com/99796443/224505134-6b8d47ec-05a0-4db1-94b6-aeb56f2f8896.png)

**Comment by IceLuna** [comment]
Sorry, but I can't seem to understand it and get it right. I've tried playing with `Demo->Examples->Dockspace` and `Demo->Examples->Documents` and couldn't get what I want. This is what I have right now:

![image](https://user-images.githubusercontent.com/99796443/224504941-be7b8e74-2756-4a9c-867a-d097ee9cbc5e.png)

```
ImGui::Begin("Tab/Parent window");
ImGui::DockSpace(dockspace_id1, ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_KeepAliveOnly);
ImGui::End();
		
ImGui::Begin("MyViewport");
ImGui::DockSpace(dockspace_id2, ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_KeepAliveOnly);
ImGui::End();
		
ImGui::Begin("Details");
ImGui::DockSpace(dockspace_id3, ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_KeepAliveOnly);
ImGui::End();
```

My goal result is something like this
![image2](https://user-images.githubusercontent.com/99796443/224505134-6b8d47ec-05a0-4db1-94b6-aeb56f2f8896.png)
